src/safeds/data/tabular/plotting/_table_plotter.py

```diff
@@ -177,55 +219,80 @@ def histograms(self, *, max_bin_count: int = 10) -> Image:
+
+            n_cols = min(3, self._table.column_count)
+            n_rows = 1 + (self._table.column_count - 1) // n_cols
 
-        for i in range(len(col_names), n_rows * n_cols):
-            fig.delaxes(axs.flatten()[i])  # Remove empty subplots
+            if n_cols == 1 and n_rows == 1:
+                fig, axs = plt.subplots(1, 1, tight_layout=True)
+                one_col = True
+            else:
+                fig, axs = plt.subplots(n_rows, n_cols, tight_layout=True, figsize=(n_cols * 3, n_rows * 3))
+                one_col = False
+
+            col_names = self._table.column_names
+            for col_name, ax in zip(col_names, axs.flatten() if not one_col else [axs], strict=False):
+                column = self._table.get_column(col_name)
+                distinct_values = column.get_distinct_values()
+
+                ax.set_title(col_name)
+                ax.set_xlabel("")
+                ax.set_ylabel("")
+
+                if column.is_numeric and len(distinct_values) > max_bin_count:
+                    min_val = (column.min() or 0) - 1e-6  # Otherwise the minimum value is not included in the first bin
+                    max_val = column.max() or 0
+                    bin_count = min(max_bin_count, len(distinct_values))
+                    bins = [
+                        *(pl.Series(range(bin_count + 1)) / bin_count * (max_val - min_val) + min_val),
+                    ]
+
+                    bars = [f"{round((bins[i] + bins[i + 1]) / 2, 2)}" for i in range(len(bins) - 1)]
+                    hist = column._series.hist(bins=bins).slice(1, length=max_bin_count).get_column("count").to_numpy()
+
+                    ax.bar(bars, hist, edgecolor="black")
+                    ax.set_xticks(range(len(hist)), bars, rotation=45, horizontalalignment="right")
+                else:
+                    value_counts = (
+                        column._series.drop_nulls().value_counts().sort(column.name).slice(0, length=max_bin_count)
+                    )
+                    distinct_values = value_counts.get_column(column.name).cast(pl.String).to_numpy()
+                    hist = value_counts.get_column("count").to_numpy()
+                    ax.bar(distinct_values, hist, edgecolor="black")
+                    ax.set_xticks(
+                        range(len(distinct_values)),
+                        distinct_values,
+                        rotation=45,
+                        horizontalalignment="right",
+                    )
+
+            for i in range(len(col_names), n_rows * n_cols):
+                fig.delaxes(axs.flatten()[i])  # Remove empty subplots
 
-        return _figure_to_image(fig)
+            return _figure_to_image(fig)
 
-    def line_plot(self, x_name: str, y_names: list[str], show_confidence_interval: bool = True) -> Image:
+    def line_plot(
+        self,
+        x_name: str,
+        y_names: list[str],
+        show_confidence_interval: bool = True,
+        *,
```

Comment:
```suggestion
        *,
        show_confidence_interval: bool = True,
```

Should also be keyword-only, was wrong before.